biorbd_optim/misc/optimal_control_program.py

```diff
@@ -615,6 +561,84 @@ def update_parameters(self, new_parameters):
+                )
+            self.__add_to_nlp("U_bounds", U_bounds, False)
+
+        if X_bounds is not None or U_bounds is not None:
+            for i in range(self.nb_phases):
+                self.__initialize_nlp(self.nlp[i])
+                Problem.initialize(self, self.nlp[i])
+            for i in range(self.nb_phases):
+                self.nlp[i].X_bounds.check_and_adjust_dimensions(self.nlp[i].nx, self.nlp[i].ns)
+                if self.nlp[i].control_type == ControlType.CONSTANT:
+                    self.nlp[i].U_bounds.check_and_adjust_dimensions(self.nlp[i].nu, self.nlp[i].ns - 1)
+                elif self.nlp[i].control_type == ControlType.LINEAR_CONTINUOUS:
+                    self.nlp[i].U_bounds.check_and_adjust_dimensions(self.nlp[i].nu, self.nlp[i].ns)
+                else:
+                    raise NotImplementedError(f"Plotting {self.nlp[i]['control_type']} is not implemented yet")
+
+        if X_init is not None:
+            if isinstance(X_init, InitialConditionsOption):
+                X_init_tp = InitialConditionsList()
+                X_init_tp.add(X_init)
+                X_init = X_init_tp
+            elif not isinstance(X_init, InitialConditionsList):
+                raise RuntimeError("X_init should be built from a InitialConditionsOption or InitialConditionsList")
+            elif len(X_init) == 0:
+                X_init.add(InitialConditionsOption(np.zeros((self.nlp[0].model.nbQ() * 2, 1))))
+            self.__add_to_nlp("X_init", X_init, False)
+
+        if U_init is not None:
+            if isinstance(U_init, InitialConditionsOption):
+                U_init_tp = InitialConditionsList()
+                U_init_tp.add(U_init)
+                U_init = U_init_tp
+            elif not isinstance(U_init, InitialConditionsList):
+                raise RuntimeError("U_init should be built from a InitialConditionsOption or InitialConditionsList")
+            elif len(U_init) == 0:
+                U_init.add(InitialConditionsOption(np.zeros((self.nlp[0].model.nbQ(), 1))))
+            self.__add_to_nlp("U_init", U_init, False)
+
+        if X_init is not None or U_init is not None:
+            for i in range(self.nb_phases):
+                self.nlp[i].X_init.check_and_adjust_dimensions(self.nlp[i].nx, self.nlp[i].ns)
+                if self.nlp[i].control_type == ControlType.CONSTANT:
+                    self.nlp[i].U_init.check_and_adjust_dimensions(self.nlp[i].nu, self.nlp[i].ns - 1)
+                elif self.nlp[i].control_type == ControlType.LINEAR_CONTINUOUS:
+                    self.nlp[i].U_init.check_and_adjust_dimensions(self.nlp[i].nu, self.nlp[i].ns)
+                else:
+                    raise NotImplementedError(f"Plotting {self.nlp[i]['control_type']} is not implemented yet")
+
+            # Variables and constraint for the optimization program
+            for i in range(self.nb_phases):
+                self.__define_multiple_shooting_nodes_per_phase(self.nlp[i], i)
```

Comment:
Probably this should be a private function by itself that can be called from __init__, update_initial_guess or update_bounds